ui/push-health/Health.jsx

```diff
@@ -114,6 +124,24 @@ export default class Health extends React.PureComponent {
     });
   };
 
+  filter = searchStr => {
+    const { location } = this.props;
+    const params = new URLSearchParams(location.search);
+
+    if (searchStr.length) {
+      params.set('searchStr', searchStr);
+    } else {
+      params.delete('searchStr');
+    }
+
+    window.history.replaceState(
```

Comment:
I'm curious if doing it this way is superior to using `props.history.replace`? There's a `updateQueryParams` helper that does it using props, btw.

Reply:
Also, I saw you comment about debouncing the search. It would result in less render-thrashing. There's a debounced input filter in the Perfherder directory called `InputFilter` if that would be useful.

Reply:
Oh cool!  OK, I moved that component to ``/shared`` and used it.  Works great!  I just made a very slight mod to it.

Reply:
Oh, and yes.  I took your suggestion about ``props.history`` and that helper function.  Good catch!